spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/data/SparkParquetWriters.java

```diff
@@ -176,6 +180,9 @@ public ParquetValueWriter<?> primitive(DataType sType, PrimitiveType primitive)
       switch (primitive.getPrimitiveTypeName()) {
         case FIXED_LEN_BYTE_ARRAY:
         case BINARY:
+          if (LogicalTypeAnnotation.uuidType().equals(primitive.getLogicalTypeAnnotation())) {
```

Comment:
The logical type annotation will be set by Parquet conversion?

Reply:
yes correct, it is being set in https://github.com/apache/iceberg/blob/master/parquet/src/main/java/org/apache/iceberg/parquet/TypeToMessageType.java#L186